src/software/ai/hl/stp/tactic/tactic.h

```diff
@@ -121,10 +151,12 @@ class Tactic
     virtual ~Tactic() = default;
 
    protected:
+    // TODO (#1888): remove this field
     // The robot performing this Tactic
-    std::optional<Robot> robot;
+    std::optional<Robot> robot_;
```

Comment:
As a general comment, I would strongly recommend using `.value()` when you're accessing `std::optional` values instead of `*` or `->` (I'm sure a past me is guilty of doing this in the codebase). This is because `.value()` actually checks the optional is valid, so you immediately get a segfault with a (hopefully) comprehensible error, whereas using the other two options will not check anything, and so the error might not show up until much later.

Reply:
If performance is a concern, it's probably worth profiling this, I suspect many of the checks will be optimized out, and even without that, the overhead is incredibly negligible with some branch prediction.

Reply:
made https://github.com/UBC-Thunderbots/Software/issues/1911 to track this issue